### PR TITLE
Sharepoint dataload schema

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,9 @@ Three schemas are used to define the metadata
 2. [closure schema](#closure-schema)
 3. [relationship schema](metadata-schema/relationshipSchema.schema.json)
 
+Additional schemas are used to define specific uses cases for metadata
```
Comment (vimleshtna) [inline README.md:37]: Rewording slightly `... specific uses cases for metadata` to `... specific use cases for metadata`

File: metadata-schema/dataLoadSharePointSchema.schema.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "$id": "/schema/dataLoadSharePointSchema",
+  "type": "object",
+  "allOf": [
+    { "properties": {
```
Comment (vimleshtna) [inline metadata-schema/dataLoadSharePointSchema.schema.json:5]: Can you please reformat it? You can use Intellij formatting `Ctrl + Alt + L`

File: src/test/resources/metadata-schema/dataLoadSharePointSchema.schema.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "$id": "/schema/dataLoadSharePointSchema",
+  "type": "object",
+  "allOf": [
+    { "properties": {
```
Comment (vimleshtna) [inline src/test/resources/metadata-schema/dataLoadSharePointSchema.schema.json:5]: Can you reformat this one, please?

File: src/test/scala/uk/gov/tna/metadata/schema/validator/SharepointDataLoadSchemaSpec.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+class SharepointDataLoadSchemaSpec extends BaseSpec {
+
+  "sharepoint data load schema validation" should {
+    "pass when all required properties present" in {
+      val schemaPath = "metadata-schema/dataLoadSharePointSchema.schema.json"
+      val dataPath = "/data/sharepointDataLoad.json"
+      val schemaInputStream = Files.newInputStream(new File(schemaPath).toPath)
+      val schema = getJsonSchemaFromStreamContentV7(schemaInputStream)
+      val dataInputStream = getClass.getResourceAsStream(dataPath)
+      val node = getJsonNodeFromStreamContent(dataInputStream)
```
Comment (vimleshtna) [inline src/test/scala/uk/gov/tna/metadata/schema/validator/SharepointDataLoadSchemaSpec.scala:21]: Can you move the above common statements to a separate method?

  ```
private def createSchema(dataPath: String) = {
    val schemaPath = "metadata-schema/dataLoadSharePointSchema.schema.json"
    val schemaInputStream = Files.newInputStream(new File(schemaPath).toPath)
    val schema = getJsonSchemaFromStreamContentV7(schemaInputStream)
    val dataInputStream = getClass.getResourceAsStream(dataPath)
    val node = getJsonNodeFromStreamContent(dataInputStream)
    (schema, node)
  }
```
And Call it from you all test methods:

      `val (schema: JsonSchema, node: JsonNode) = createSchema(dataPath)
`

**Review by vimleshtna — COMMENTED** [review]
Looks good. Just a few minor comments. Thanks

File: src/test/scala/uk/gov/tna/metadata/schema/validator/SharepointDataLoadSchemaSpec.scala (hunk 4 above)
Comment (TomJKing) [reply src/test/scala/uk/gov/tna/metadata/schema/validator/SharepointDataLoadSchemaSpec.scala]: Good point.

**Review by vimleshtna — APPROVED** [review]
Looks good. Thank for doing this,

**Pull request merged**